src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -48,27 +48,23 @@ public class UserManager<TUser> : IDisposable where TUser : class
         /// <param name="contextAccessor">The accessor used to access the <see cref="HttpContext"/>.</param>
         public UserManager(IUserStore<TUser> store,
             IOptions<IdentityOptions> optionsAccessor,
-            IPasswordHasher<TUser> passwordHasher,
-            IEnumerable<IUserValidator<TUser>> userValidators,
-            IEnumerable<IPasswordValidator<TUser>> passwordValidators,
-            ILookupNormalizer keyNormalizer,
-            IdentityErrorDescriber errors,
             IServiceProvider services,
-            ILogger<UserManager<TUser>> logger,
-            IHttpContextAccessor contextAccessor)
+            ILogger<UserManager<TUser>> logger)
         {
             if (store == null)
             {
                 throw new ArgumentNullException(nameof(store));
             }
             Store = store;
-            Options = optionsAccessor?.Value ?? new IdentityOptions();
-            _context = contextAccessor?.HttpContext;
-            PasswordHasher = passwordHasher;
-            KeyNormalizer = keyNormalizer;
-            ErrorDescriber = errors;
             Logger = logger;
+            Options = optionsAccessor?.Value ?? new IdentityOptions();
+
+            _context = services?.GetService<IHttpContextAccessor>()?.HttpContext;
+            PasswordHasher = services?.GetService<IPasswordHasher<TUser>>() ?? new PasswordHasher<TUser>();
+            KeyNormalizer = services?.GetService<ILookupNormalizer>() ?? new UpperInvariantLookupNormalizer(); ;
```

Comment:
:+1: to everything @halter73 said. 

Besides the lack of clarity on the actual dependencies and ambiguous optionality in the constructor, from the code maintainability perspective this adds a new alternate source of truth about our dependencies. Rather than having it all encoded in `AddIdentity()` & friends, it is fragmented between that and this constructor.

Ultimately I agree that this is about how much we care about making it easy to create a UserManager without using DI, and I personally don't think it is a goal.